MennaMohie/assignment2/Solution.cpp

```diff
@@ -152,22 +147,40 @@ class BinaryTree
 int main()
 {
     BinaryTree<int> tree(1);
-    tree.addAsLeftChild(tree.getRoot(), 1, 2);
-    tree.addAsRightChild(tree.getRoot(), 1, 3);
-    tree.addAsLeftChild(tree.getRoot(), 2, 4);
-    tree.addAsRightChild(tree.getRoot(), 2, 5);
-    tree.addAsLeftChild(tree.getRoot(), 3, 6);
-    tree.addAsRightChild(tree.getRoot(), 3, 7);
-    tree.addAsLeftChild(tree.getRoot(), 7, 8);
-    tree.addAsRightChild(tree.getRoot(), 8, 9);
-
-    tree.printAncestors(tree.getRoot(), 9);
-    cout<<endl;
+    tree.addChildToParent(tree.getRoot(), 1, 2, addLeft);
```

Comment:
It's kind of a strange interface when you are calling a method on the tree and still need to pass the root to it. It kind of knows it's own root :)